### PR TITLE
PRSD-373: Landlord Registration Declaration

### PR DESCRIPTION
**Pull request by isobel-softwire**
### Overarching Changes
- Creates checkboxes component

### Features
- Adds `Declaration` step to `LandlordRegistrationJourney`

### Tests
- Integration tests for the `Declaration` step

### Screenshot
Declaration page
image

Declaration page with error message
image

### CODE REVIEW

**Comment by JasminConterioSW** [comment]
If you get a chance, it can be useful to add a screenshot of the new page when it's a UI change.

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -25,6 +25,9 @@ spring:
 
   mvc.format.date: "d MMMM yyyy"
 
+  thymeleaf:
+    render-hidden-markers-before-checkboxes: true
```
[the remark this answers is not in the document]
Comment (Travis-Softwire) [reply src/main/resources/application.yml:29]: Intriguing! Would be good to understand what this does offline (but don't let it hold up this PR)

**Comment by isobel-softwire** [comment]
> If you get a chance, it can be useful to add a screenshot of the new page when it's a UI change.

Forgot to do this! Added some now :)

File: src/main/kotlin/uk/gov/communities/prsdb/webapp/models/formModels/DeclarationFormModel.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+class DeclarationFormModel : FormModel {
+    @ValidatedBy(
+        constraints = [
+            ConstraintDescriptor(
+                messageKey = "forms.declaration.checkbox.error.missing",
+                validatorType = TrueConstraintValidator::class,
+            ),
+        ],
+    )
+    var agreesToDeclaration: Boolean? = false
```
[the remark this answers is not in the document]
Comment (isobel-softwire) [reply src/main/kotlin/uk/gov/communities/prsdb/webapp/models/formModels/DeclarationFormModel.kt]: It doesn't - changed this now!

**Comment by JasminConterioSW** [comment]
On Figma the warning text isn't bold, but I think your version is correct (on GDS the warning text **is** bold). It might be worth dropping a message to designers to ask if they can make it bold to match?

**Review by JasminConterioSW — APPROVED** [review]
LGMT :)

**Pull request merged**